src/python/pants/pantsd/watchman.py

```diff
@@ -147,8 +149,12 @@ def watch_project(self, path):
     try:
       return self.client.query('watch-project', os.path.realpath(path))
     finally:
-      self.client.setTimeout(self._timeout)
-      self._logger.debug('setting post-startup watchman timeout to %s', self._timeout)
+      try:
```

Comment:
nested try-catches not terribly clear... maybe break into another method?

Reply:
done